sdks/java/core/src/main/java/org/apache/beam/sdk/testing/LargeKeys.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.testing;
+
+/**
+ * Category tags for tests which validate that a Beam runner can handle keys up to a given size.
+ */
+public interface LargeKeys {
+  /**
+   * Tests if a runner supports 10KB keys.
+   */
+  public interface UpTo10KB {}
```

Comment:
I am getting my head turned inside out because of how these names interact with negation.

 - `<groups>UpToXYZ</groups>` will include all tests of keys sizes up to XYZ
 - `<excludedGroups>UpToXYZ</excludedGroups>` will exclude key sizes above XYZ, somewhat the opposite of the reading

I'm not sure there is a way to solve this reading issue...

Incidentally, you can use inheritance between the interfaces. I believe they should be set up to favor exclusion - presumably a runner just wants to exclude keys above a certain threshold. So probably `<excludedGroups>LargeKeys.AboveXYZ</excludedGroups>` is the right way to go, and no need to list multiple.